### PR TITLE
Added seamless router reload

### PR DESCRIPTION
**Pull request by jkroepke**
Configure the seamless router reload like described here:
https://docs.openshift.com/container-platform/3.9/install_config/router/default_haproxy_router.html#preventing-connection-failures-during-restarts

Tested in a running 3.9 cluster.

### CODE REVIEW

**Comment by vrutkovs** [comment]
That seems to work great here, lets see if Scott or Michael have an opinion on this

**Comment by jkroepke** [comment]
It might be deprecated: https://github.com/openshift/origin/pull/19315

Docs are not updated :-(

File: roles/openshift_hosted/tasks/router.yml [hunk 1]
```diff
@@ -73,6 +73,20 @@
...
+    user: "system:serviceaccount:{{ item.namespace }}:{{ item.serviceaccount }}"
+    namespace: "{{ item.namespace }}"
+    resource_kind: scc
+    resource_name: privileged
+    state: "{{ 'present' if openshift_hosted_router_seamless_reload else 'absent' }}"
+  with_items: "{{ openshift_hosted_routers }}"
+
+- name: Configure seamless router restart
+  set_fact:
+    openshift_hosted_router_edits: "{{ openshift_hosted_router_seamless_edits | union(openshift_hosted_router_edits) }}"
```
Comment (michaelgugino) [inline roles/openshift_hosted/tasks/router.yml]: Don't use set_fact here.  It's not a good practice to set_fact an existing variable name.

Comment (michaelgugino) [reply roles/openshift_hosted/tasks/router.yml]: You can apply the following to "Create OpenShift router" task:

```yaml
  edits: "{{ l_openshift_hosted_router_edits | union(item.edits) }}"
vars:
  l_ohr_seemless_edits: "{{ (openshift_hosted_router_seamless_reload | bool) | ternary(openshift_hosted_router_seamless_edits, []) }}"
  l_openshift_hosted_router_edits: "{{ openshift_hosted_router_edits | union(l_ohr_seemless_edits) }}"
```

**Comment by ironcladlou** [comment]
Too late in the release to merge this, and we also need to reconcile it with our 4.0 design plans. Going to hold it for now. In the meantime, @ramr @pravisankar @knobunc PTAL

**Comment by pravisankar** [comment]
Created 3.9 backport PR for openshift/origin#19315 -->  https://github.com/openshift/ose/pull/1390

**Comment by ironcladlou** [comment]
Closing as obsolete per https://github.com/openshift/openshift-ansible/pull/9546#issuecomment-412977233. Thanks for bringing it to our attention, @jkroepke! Sorry to close the PR, your effort is greatly appreciated.

**Pull request closed without merging**